### PR TITLE
- Add Direction Feature when mode is a modal

### PR DESCRIPTION
**Pull request by Kottidev**
- Default Direction vertical

- other values direction (vertical ,horizontal, leftToRight ,topToBottom ,none, zoomIn)

### CODE REVIEW

**Comment by nihgwu** [comment]
a **fade** more would be much better, which is one of the animationType of `Modal`

**Comment by matthamil** [comment]
When submitting or proposing new features, please update the documentation to allow users to easily implement the feature and help project maintainers understand your goals for the proposed feature. Thanks for contributing!

**Comment by Kottidev** [comment]
@matthamil sorry i forget to update documentation

File: docs/api/navigators/StackNavigator.md [hunk 1]
```diff
@@ -79,7 +79,14 @@ Visual options:
 
 - `mode` - Defines the style for rendering and transitions:
   - `card` - Use the standard iOS and Android screen transitions. This is the default.
-  - `modal` - Make the screens slide in from the bottom which is a common iOS pattern. Only works on iOS, has no effect on Android.
+  - `modal` - Make the screens slide in from the bottom.
+- `direction` - direction of animation when mode is modal
+  - `vertical` - Use the standard Android screen transitions, will be bottom to top. This is the default.
+  - `horizontal` - will be right to left .
```
Comment (matthamil) [inline docs/api/navigators/StackNavigator.md]: nit: remove space before period

Comment (matthamil) [inline docs/api/navigators/StackNavigator.md]: nit: Can you include more verbiage and/or be more descriptive than "will be right to left"? This is particularly important for those who might not be very familiar with English.

**Comment by drop-dan** [comment]
Would love to see this feature merged before 1.0! Is it just documentation preventing the merge?

**Comment by ericvicenti** [comment]
This looks like a reasonable approach, but I'm not sure it will solve a lot of people's requests. Instead of configuring the whole navigator, I've noticed that a lot of people are requesting to configure one screen at a time.

**Comment by nihgwu** [comment]
@ericvicenti I think this it reasonable no matter it's for the stack or screen, the workaround for screen modal is actually a stack too, right now I have to implement my own modal navigator for each project only for a fade transition

**Comment by grabbou** [comment]
I agree with @ericvicenti that it should work on per-screen basis, setting animation type on entire navigator doesn't solve much of my pains.

Also, PR needs to be rebased.

**Comment by Kottidev** [comment]
I will rebuilt it with per-screen ;)

@ericvicenti @grabbou

**Comment by RWOverdijk** [comment]
👍 Looking for this as I'm currently implementing a filter modal.

**Comment by superandrew213** [comment]
Any news on this?

**Comment by kelset** [comment]
Looks like this PR currently needs some revision, there are conflicting files.

**Comment by brentvatne** [comment]
thank you for opening this PR! at the moment we are in the process of re-thinking how we do modals in react-navigation. you can comment on https://github.com/react-navigation/rfcs/issues/15 with ideas or open up a RFC if you have a plan for how you would like this to work!

unfortunately we can't ship the code in this PR in its current form because it doesn't merge cleanly with master. I apologize that there wasn't enough follow-up on this PR to begin with, we're trying to do better at handling pull requests in a more timely manner going forward. apologies to anybody who hoped this would land -- please feel free to open a feature request on http://react-navigation.canny.io/ or participate in the RFC if this is important to you!

**Pull request closed without merging**